esx_service/utils/auth_data.py

```diff
@@ -0,0 +1,568 @@
+                
+                # search privileges and default_privileges for this tenant
+                privileges = []
+                cur = self.conn.execute(
+                    "SELECT * FROM privileges WHERE tenant_id = ? AND datastore != ?",
+                    (id,default_datastore)    
+                )
+                privileges = cur.fetchall()
+
+                default_privileges = []
+                cur = self.conn.execute(
+                    "SELECT * FROM privileges WHERE tenant_id = ? AND datastore = ?",
+                    (id,default_datastore)    
+                )
+                default_privileges = cur.fetchall()
+                tenant = DockerVolumeTenant(name, description, default_datastore,
+                                            default_privileges, vms, privileges, id)
+                tenant_list.append(tenant)
+        except sqlite3.Error, e:    
+            logging.error("Error %s when listting all tenants", e)
+
+        return tenant_list   
+    
+    def remove_volumes_from_volume_table(self, tenant_id):
+        """ Remove all volumes from volumes table. """
+        try:
+            self.conn.execute(
+                    "DELETE FROM volumes WHERE tenant_id = ?",  
+                    [tenant_id]
+            )
+
+            self.conn.commit()
+        except sqlite3.Error, e:
+            logging.error("Error %s when removing volumes from volumes table for tenant_id %s",
+                          e, tenant_id)
+    
+    def _remove_volumes_for_tenant(self, tenant_id):
+        """ Delete all volumes belongs to this tenant.
+
+            Do not use it outside of removing a tenant.
+
+        """
+        try:
+            cur = self.conn.execute(
+            "SELECT name FROM tenants WHERE id = ?",
+            (tenant_id,)
+            )
+            result = cur.fetchone()
+        except sqlite3.Error, e:
+            logging.error("Error %s when querying from tenants table", e)
+
+        logging.debug("remove_volumes_for_tenant: %s %s", tenant_id, result)
+        tenant_name = result[0]
+        vmdks = vmdk_utils.get_volumes(tenant_name)
+        # Delete all volumes for this tenant. 
+        dir_path = None
+        for vmdk in vmdks:
+            vmdk_path = os.path.join(vmdk['path'], "{0}".format(vmdk['filename']))
+            if not dir_path:
+                dir_path = vmdk['path']
+            print("path=%s filename=%s", vmdk['path'], vmdk['filename'])    
+            logging.debug("Deleting volume path%s", vmdk_path)
+            err = vmdk_ops.removeVMDK(vmdk_path)
+            if err:
+                logging.error("remove vmdk %s failed with error %s", vmdk_path, err)
+        
+        # Delete path /vmfs/volumes/datastore_name/tenant_name
+        logging.debug("Deleting dir path%s", dir_path)
+        try:
+            os.rmdir(dir_path)
+        except os.error as e:
+            logging.error("remove dir %s failed with error %s", dir_path, e)
```

Comment:
Add code to log and return  the error when actually deleting the volumes. If any error happens, stop and not delete them from tables.